tpm2/test/audit_test.go

```diff
@@ -25,38 +25,39 @@ func TestAuditSession(t *testing.T) {
 	// Create the AK for audit
 	createAKCmd := CreatePrimary{
 		PrimaryHandle: TPMRHOwner,
-		InPublic: TPM2BPublic{
-			PublicArea: TPMTPublic{
-				Type:    TPMAlgECC,
-				NameAlg: TPMAlgSHA256,
-				ObjectAttributes: TPMAObject{
-					FixedTPM:             true,
-					STClear:              false,
-					FixedParent:          true,
-					SensitiveDataOrigin:  true,
-					UserWithAuth:         true,
-					AdminWithPolicy:      false,
-					NoDA:                 true,
-					EncryptedDuplication: false,
-					Restricted:           true,
-					Decrypt:              false,
-					SignEncrypt:          true,
-				},
-				Parameters: TPMUPublicParms{
-					ECCDetail: &TPMSECCParms{
-						Scheme: TPMTECCScheme{
-							Scheme: TPMAlgECDSA,
-							Details: TPMUAsymScheme{
-								ECDSA: &TPMSSigSchemeECDSA{
-									HashAlg: TPMAlgSHA256,
-								},
+		InPublic: TPM2BPublic(&TPMTPublic{
+			Type:    TPMAlgECC,
+			NameAlg: TPMAlgSHA256,
+			ObjectAttributes: TPMAObject{
+				FixedTPM:             true,
+				STClear:              false,
+				FixedParent:          true,
+				SensitiveDataOrigin:  true,
+				UserWithAuth:         true,
+				AdminWithPolicy:      false,
+				NoDA:                 true,
+				EncryptedDuplication: false,
+				Restricted:           true,
+				Decrypt:              false,
+				SignEncrypt:          true,
+			},
+			Parameters: TPMUPublicParms(
+				TPMAlgECC,
```

Comment:
Little bit irritating to use here, can we avoid the user having to pass the selector

Reply:
Consider making the underlying tpmu contain an interface that happens to be implemented (in dummy methods)

Reply:
Unfortunately, the selector cannot always be inferred from the type of the contents, because some TPU contain multiple different members that have the same type.